### PR TITLE
Bluetooth: Mesh: Friendship additions, cleanups & fixes

### PR DESCRIPTION
**Pull request by jhedberg**
The biggest change in this PR is the addition of Friend support (the first patch in the set). The rest are either Friend or LPN related fixes or improvements. The patches have received plenty of testing with the PTS, so I consider the functionality "good enough" to be merged. It's possible there are still things to fix, but I'd rather do those in follow-up patches.

### CODE REVIEW

**Comment by jhedberg** [comment]
@michal-narajowski FYI

File: subsys/bluetooth/host/mesh/net.c [hunk 1]
```diff
@@ -1167,7 +1167,7 @@ void bt_mesh_net_recv(struct net_buf_simple *data, s8_t rssi,
 		      enum bt_mesh_net_if net_if)
 {
 	struct net_buf_simple *buf = NET_BUF_SIMPLE(29);
-	struct bt_mesh_net_rx rx;
+	struct bt_mesh_net_rx rx = { .rssi = rssi };
```
Comment (Vudentz) [inline subsys/bluetooth/host/mesh/net.c]: I guess the commit should say Fix not including...

Comment (jhedberg) [reply subsys/bluetooth/host/mesh/net.c]: Well, it depends how you read it. If you read it as "Fix <feature>" and the <feature> is "including RSSI in the RX context" then the current commit message makes sense.

Comment (Vudentz) [reply subsys/bluetooth/host/mesh/net.c]: Right maybe a description then, anyway other than that is looks fine.

Comment (jhedberg) [reply subsys/bluetooth/host/mesh/net.c]: I changed it to the form that you were expecting, and added an extra sentence to the commit message body as well.

**Review by Vudentz — APPROVED** [review]
LGTM

**Pull request merged**